includes/class-newspack-popups-inserter.php

```diff
@@ -184,89 +199,77 @@ public static function insert_popup( $content = '', $popup = [] ) {
 		}
 		$before_popup = substr( $content, 0, $insertion_position );
 		$after_popup  = substr( $content, $insertion_position );
-		return $before_popup . $popup['markup'] . $after_popup;
+
+		return $before_popup . $popup_markup . $after_popup;
 	}
 
 	/**
-	 * Insert inline Popup markup into content
+	 * The popup shortcode function.
 	 *
-	 * @param string $content The content of the post.
-	 * @param object $popup The popup object to insert.
-	 * @return string The content with popup inserted.
+	 * @param array $atts Shortcode attributes.
+	 * @return HTML
 	 */
-	public static function insert_inline_popup( $content = '', $popup = [] ) {
-		$position  = 0;
-		$positions = [];
-		$close_tag = '</p>';
-		while ( stripos( $content, $close_tag, $position ) !== false ) {
-			$position    = stripos( $content, '</p>', $position ) + strlen( $close_tag );
-			$positions[] = $position;
-		}
-		$total_length       = strlen( $content );
-		$percentage         = intval( $popup['options']['trigger_scroll_progress'] ) / 100;
-		$precise_position   = $total_length * $percentage;
-		$insertion_position = $total_length;
-		foreach ( $positions as $position ) {
-			if ( $position >= $precise_position ) {
-				$insertion_position = $position;
-				break;
-			}
+	public static function popup_shortcode( $atts = array() ) {
+		$previewed_popup_id = Newspack_Popups::previewed_popup_id();
+		if ($previewed_popup_id) {
+			return Newspack_Popups_Model::retrieve_preview_popup( $previewed_popup_id )['markup'];
+		} else {
+			return Newspack_Popups_Model::retrieve_popup_by_id( $atts['id'] )['markup'];
```

Comment:
Should this have a check that `$atts['id']` exists, and that the pop-up object returned actually has a `markup` param?

Reply:
Fixed: 829adc938292df92b9347d2b137d3530cb5635d2